### PR TITLE
[LETS-468] Active Transaction Server seems to reset mvccid with each restart

### PR DESCRIPTION
**Pull request by adrianbrab**
http://jira.cubrid.org/browse/LETS-468

The `MVCCID` seems to reset after the ATS closes. To patch adds the possibility for the `MVCCID` to be set and updated.

### CODE REVIEW

File: src/transaction/log_replication.cpp [hunk 1]
```diff
@@ -338,6 +338,7 @@ namespace cublog
     log_replication_update_header_mvcc_vacuum_info (mvccid, rec_header.back_lsa, rec_lsa, m_bookkeep_mvcc_vacuum_info);
     if (m_replicate_mvcc && MVCCID_IS_NORMAL (mvccid))
       {
+	log_Gl.mvcc_table.set_mvccid_from_active_transaction_server (mvccid);
```
Comment (cristiarg) [inline src/transaction/log_replication.cpp:351]: implementation missing in function `register_assigned_mvccid`

File: src/transaction/mvcc_table.cpp [hunk 2]
```diff
@@ -587,6 +588,21 @@ mvcctable::get_two_new_mvccid (MVCCID &first, MVCCID &second)
...
+mvcctable::set_mvccid_from_active_transaction_server (MVCCID id)
+{
+  // note that function is called from multiple replication threads/tasks/daemons
+
+  assert (is_page_server ());
+  m_new_mvccid_lock.lock ();
+  // only incremental, never backwards
+  if (id >= log_Gl.hdr.mvcc_next_id)
+    {
+      log_Gl.hdr.mvcc_next_id = id + 1;
```
Comment (cristiarg) [inline src/transaction/mvcc_table.cpp]: use `MVCCID_FORWARD`

File: src/transaction/mvcc_table.hpp [hunk 3]
```diff
@@ -81,6 +81,7 @@ class mvcctable
     void complete_sub_mvcc (MVCCID mvccid);
     MVCCID get_new_mvccid ();
     void get_two_new_mvccid (MVCCID &first, MVCCID &second);
+    void set_mvccid_from_active_transaction_server (MVCCID id);
```
Comment (cristiarg) [inline src/transaction/mvcc_table.hpp:85]: explanation for function

File: src/transaction/mvcc_table.cpp [hunk 4]
```diff
@@ -587,6 +588,21 @@ mvcctable::get_two_new_mvccid (MVCCID &first, MVCCID &second)
   m_new_mvccid_lock.unlock ();
 }
 
+void
+mvcctable::set_mvccid_from_active_transaction_server (MVCCID id)
+{
+  // note that function is called from multiple replication threads/tasks/daemons
+
+  assert (is_page_server ());
```
Comment (cristiarg) [inline src/transaction/mvcc_table.cpp:596]: this assert contradicts the calling place condition `if (m_replicate_mvcc &&..`

File: src/transaction/log_replication.cpp (hunk 1 above)
Comment (adrianbrab) [reply src/transaction/log_replication.cpp:351]: isn't the function `register_assigned_mvccid` only called on PTS?

**Review by cristiarg — APPROVED** [review]

**Pull request merged**